pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java

```diff
@@ -67,7 +102,26 @@ static public void start(Context ctx, String pythonServiceArgument) {
         intent.putExtra("pythonHome", argument);
         intent.putExtra("pythonPath", argument + ":" + argument + "/lib");
         intent.putExtra("pythonServiceArgument", pythonServiceArgument);
-        ctx.startService(intent);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            ctx.startForegroundService(intent);
+        } else {
+            ctx.startService(intent);
+        }
+    }
+
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+    }
+
+    /**
+     * Stops the task gracefully when killed.
+     * Calling stopSelf() will trigger a onDestroy() call from the system.
```

Comment:
Misleading comment, also why do we override method if we don't add/change anything?

Reply:
Good point :) 
:+1: